pkg/volume/volume.go

```diff
@@ -151,24 +151,30 @@ func getVolumeClassNames(volumeClasses map[string]adminv1.ProjectVolumeClassInst
 	return typed.SortedKeys(storageClassName)
 }
 
-func CopyVolumeDefaults(volumeRequest v1.VolumeRequest, volumeBinding v1.VolumeBinding, volumeDefaults v1.VolumeDefault) v1.VolumeRequest {
+func ResolveVolumeRequest(volumeRequest v1.VolumeRequest, volumeBinding v1.VolumeBinding, resolvedVolume v1.VolumeResolvedOffering) v1.VolumeRequest {
+	// Order of precedence:
+	// 1. VolumeResolvedOffering - the previously resolved values for Class, Size, and AccessModes
+	// 2. VolumeBinding - the values set by the user in the app spec
+	// 3. VolumeRequest - the defaults as defined in the acorn image
```

Comment:
Volume bindings are the ones set by the user at runtime (with the `-v` flag). I think these are the top precedence.